ign_rviz_plugins/src/rviz/plugins/AxesDisplay.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ignition/rviz/plugins/AxesDisplay.hpp"
+#include <ignition/plugin/Register.hh>
+#include <ignition/gui/Application.hh>
+#include <ignition/gui/GuiEvents.hh>
+
+#include <algorithm>
+#include <memory>
+#include <utility>
+#include <vector>
+#include <string>
+
+namespace ignition
+{
+namespace rviz
+{
+namespace plugins
+{
+////////////////////////////////////////////////////////////////////////////////
+AxesDisplay::AxesDisplay()
+: length(1.0), radius(0.1), headVisible(false), dirty(false)
+{
+  this->engine = rendering::engine("ogre");
```

Comment:
`ogre`  is hardcoded, Can we avoid this?

Reply:
We can load the rendering engine name from the config file and then propagate it to all the plugins. 
For now, I have added a TODO for this. I'll add this feature when working on the loading/saving config file.